api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1748,8 +1770,15 @@ type AzurePlatformSpec struct {
 	// +immutable
 	VnetName string `json:"vnetName"`
 
-	// VnetID is the ID of an existing VNET to use in creating VMs.
-	// In ARO HCP, this will be the ID of the customer provided VNET.
+	// VnetID is the ID of an existing VNET to use in creating VMs. If this field is included, it should
+	// match the same VNET as VnetID. This VNET is expected to exist under the same subscription as SubscriptionID. If
+	// this field is included, it should be the VNET ID of VnetName.
+	//
+	// This resource is expected to exist in VnetResourceGroupName. If VnetResourceGroupName is not provided, HyperShift
+	// will assume it is in ResourceGroupName.
```

Comment:
Same comment as https://github.com/openshift/hypershift/pull/3945/files#r1591922974

Reply:
VnetResourceGroupName

Reply:
This is OBE with the removal of VnetResourceGroupName